File: authorization-manager-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__AuthorizationManagerDescriptor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.Nullable;
import se.curity.identityserver.sdk.authorization.graphql.GraphQLAuthorizationManager;
import se.curity.identityserver.sdk.authorization.oauth.OAuthAuthorizationManager;
import se.curity.identityserver.sdk.authorization.scim.ScimAuthorizationManager;
import se.curity.identityserver.sdk.plugin.descriptor.AuthorizationManagerPluginDescriptor;

public final class ${pluginName}AuthorizationManagerDescriptor implements AuthorizationManagerPluginDescriptor<${pluginName}AuthorizationManagerConfig>
{
    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}AuthorizationManagerConfig> getConfigurationType()
    {
        return ${pluginName}AuthorizationManagerConfig.class;
    }

    @Override
    public @Nullable Class<? extends GraphQLAuthorizationManager> getGraphQLAuthorizationManager()
    {
        return ${pluginName}GraphQLAuthorizationManager.class;
    }

    @Override
    public @Nullable Class<? extends OAuthAuthorizationManager> getOAuthAuthorizationManager()
    {
        return ${pluginName}OAuthAuthorizationManager.class;
    }

    @Override
    public @Nullable Class<? extends ScimAuthorizationManager> getScimAuthorizationManager()
    {
        return ${pluginName}ScimAuthorizationManager.class;
    }
}

File: kotlin-authorization-manager-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__AuthorizationManagerDescriptor.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.plugin.descriptor.AuthorizationManagerPluginDescriptor

class ${pluginName}AuthorizationManagerDescriptor: AuthorizationManagerPluginDescriptor<${pluginName}AuthorizationManagerConfig>
{
    override fun getPluginImplementationType() = "${pluginName.toLowerCase()}"

    override fun getConfigurationType() = ${pluginName}AuthorizationManagerConfig::class.java

    override fun getGraphQLAuthorizationManager() = ${pluginName}GraphQLAuthorizationManager::class.java

    override fun getOAuthAuthorizationManager() = ${pluginName}OAuthAuthorizationManager::class.java

    override fun getScimAuthorizationManager() = ${pluginName}ScimAuthorizationManager::class.java
}
